src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/CacheClearTest.kt

```diff
@@ -0,0 +1,139 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import com.github.tomakehurst.wiremock.client.WireMock
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.ContextStaffMemberFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.OffenderDetailsSummaryFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given a User`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.httpmocks.APDeliusContext_mockSuccessfulStaffMembersCall
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.httpmocks.CommunityAPI_mockSuccessfulOffenderDetailsCall
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserRole
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.community.OffenderDetailSummary
+import java.util.UUID
+
+class CacheClearTest : IntegrationTestBase() {
+  @Test
+  fun `Clearing a regular cache results in upstream calls being made again`() {
+    `Given a User`(roles = listOf(UserRole.MANAGER)) { _, jwt ->
+      val premisesId = UUID.fromString("d687f947-ff80-431e-9c72-75f704730978")
+      val qCode = "FOUND"
+
+      setupQCodeUpstreamResponse(premisesId, qCode)
+
+      // Call our endpoint twice, first time should cache
+      repeat(2) {
+        webTestClient.get()
+          .uri("/premises/$premisesId/staff")
+          .header("Authorization", "Bearer $jwt")
+          .exchange()
+          .expectStatus()
+          .isOk
+      }
+
+      // Validate upstream request only made once, so the second request was served from cache
+      validateUpstreamQCodeRequestMade(qCode, times = 1)
+
+      // Clear the qCode cache
+      webTestClient.delete()
+        .uri("/cache/qCodeStaffMembers/clear")
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      // Call our endpoint again
+      webTestClient.get()
+        .uri("/premises/$premisesId/staff")
+        .header("Authorization", "Bearer $jwt")
+        .exchange()
+        .expectStatus()
+        .isOk
+
+      // Validate upstream request was made again, proving the cache was cleared
```

Comment:
Loving these comments - makes it so much easier to follow! Might be worth following the convention we use in the UI and use Cucumber-like syntax, ie:

- Given I call the qCode endpoint twice
- Then the upstream request should only have been made once
- When I clear the qCode cache
- And I call the endpoint again
- Then the upstream request should be made again

Reply:
Done for that one and the other test too